### DISCUSSION TITLE
Lower Requirement to Trigger BTCpay Order Match from 0.001 BTC to Something Lower

### DISCUSSION THREAD

**Discussion by davestaxcp** [CounterpartyXCP/CIPs#96] [CIPs]
Below is a test process I used, along with help from @blockjack8 and @al_fernandz to try to use the BTCpay function using Freewallet v0.9.23 and we found orders were not matching with BTC orders under 0.001 BTC.

_______________________________________________________________
Here is what we tested:

So I've been working together with @blockjack8 to develop a perfect FAQ for BTCpay for users using Freewallet v0.9.23. He and I dug into the forums, github, asked in dev chat and counterparty chat about this process. We are trying to hash this out for users to make it the most smooth process and easy for anybody new to Counterparty. Both Angel and I haven't used BTCpay before, so we both thought we would test the process.

Angel threw up a test BTC/RAREAIRDROP market and we got some help early on from @al_fernandz , in which Al successfully filled his BTCpay order here: https://xchain.io/tx/2428927

It was Al's experience that this simply "filled automatically" when he made sure to keep Auto BTCpay in the ON setting when making the BTCpay order. Which was completed and filled. He is also using v0.9.23.

Angel and I also then tried to do the same process, but our orders never automatically filled even though the market had open sell orders that should have automatically matched for us to buy with btc. It was strange because we did not have the same experience as Al.

I went even farther to try filling the order on the Auto setting (https://xchain.io/tx/2429910) and the Manual setting which I expected some sort of popup while keeping Freewallet open (https://xchain.io/tx/2429937). No fills. No popup at that time.

I took a break after many hours of nothing "automatically happening" and the next day I would eventually close and restart my Freewallet to see these two popups:

"Enable Auto-BTC pay?" - which it then asked for my password (just the one I created in Freewallet to lock my account, which is normal for unlocking Freewallet for other functions)

Then it popped up a statement "Auto BTCpay is now enabled and any order matches for BTC will be automatically paid"

So I thought, okay, why not try again? I used Auto BTC pay setting again assuming it should work now (https://xchain.io/tx/2430021). Still no luck. No order matches like Al's tx did , still not "working" from my user experience. Even stranger, Angel nor Al every saw these "Auto-BTCpay" popups.

In Angel's and my experience, BTCpay function is not working correctly for us in v0.9.23

So Al recently did a little experiment today. He tried (as Angel and I did) putting in a BUY order for buying 1 RAREAIRDROP for 0.002 BTC (as it SHOULD match the sell/buy list already open on the RAREAIRDROP/BTC market) - https://xchain.io/tx/e3e9b1ccaa1f1cc71478651e0e660eda98d52a64809190286158e3b7b686a1a1

Al also then put in a SELL order for 0.001 for 5 RAREAIRDROP to test the other direction of the trade - https://xchain.io/tx/ff6408b1da973a7f18256d391d0bc658b08d53329522f80f1e8dd884b96035a0

Now it looks like neither of the two new tx's from Al have matched correctly. So now I'm gonna stop wrapping my head around this and look into making other FAQ's for other functions. This will need to be sorted out before Angel and I can successfully write an FAQ on this subject.
__________________________________________________________________________________________

It came to my understanding (thanks @hodlencoinfield) that BTCpay requires a BTC amount of at least 0.001 BTC to match. 

This is really awkward to explain and write into an FAQ and seems to be something that should be changed so people can still use BTCpay for "small" BTC amounts.

From my user experience as well as turning new users on the Freewallet and the Counterparty platform, users in my opinion usually go to BTC/TOKEN first. And these users most likely are using small BTC amounts because they are using a new platform and testing things out.

Not only do we need a perfect and updated FAQ on how to use BTCpay, we need it to work for small amounts of BTC.

We can write in the FAQ this requirement (put orders up for at least 0.001), but I think it should be changed to the lowest amount possible. One satoshi maybe if possible?

I have seen so many people spend insane amounts for BTC fee's to buy tokens worth much less than 0.001 BTC so I'm not sure where the Dev's stand on this issue, but as a user, if you put an order in with the education presented now, it feels "broken."

**Comment by jdogresorg** [comment]
I just checked `counterparty-lib` code... we lowered the `min_btc_quantity` value to 0.00001 BTC a while ago

https://github.com/CounterpartyXCP/counterparty-lib/blob/master/counterpartylib/lib/messages/order.py#L443-L446

Here is the PR with the update
https://github.com/CounterpartyXCP/counterparty-lib/pull/1070

So... if we are stuck at 0.001 still, seems this is could be a bug.

**Reply by Jpja** [reply]
Do small orders <0.00001 BTC remain in the order book?

Should not be possible to place such small orders AND small remainders after partial matches should be deleted.

Years since I tested this but I remember DEX orders that I expected to match did not (probably because < `min_btc_quantity`) and sometimes bid > ask (because small orders not matching).

**Comment by jotapea** [comment]
Freewallet is a private product by @jdogresorg .

While I can appreciate the reason for this post was to solve a problem / bug, we need to stop mixing Freewallet / XChain as part of the Counterparty protocol. If @jdogresorg cleared up that the protocol is correct, then the follow up to this should happen in the Freewallet repo.

**Reply by jdogresorg** [reply]
Once again, I encourage you to keep your post ON TOPIC. This is a discussion about finding an issue with BTCpay and fixing it, not a post about your dislike of xchain or freewallet. 

This discussion was created by @davestaxcp using the ONLY wallet (freewallet) that works with BTCpay to test BTCpay and write up documentation to make using Counterparty easier for new users.

The issue discovered (and referenced above) was in the CP protocol, nothing to do with freewallet or xchain specific settings.

**Reply by jotapea** [reply]
Is the issue in the protocol or not?

If is not, then great and lets move on and close this discussion. And if you want to follow up on Freewallet repo, feel free to link to it from here.

**Reply by loon3** [reply]
yes it appears to be a protocol issue

**Comment by adamkrellenstein** [comment]
This sounds like a simple bugfix, albeit a protocol-level one. Tracking it here: https://github.com/CounterpartyXCP/counterparty-core/issues/1278
